sqlglot/optimizer/qualify_columns.py

```diff
@@ -22,6 +22,7 @@ def qualify_columns(
     expand_alias_refs: bool = True,
     expand_stars: bool = True,
     infer_schema: t.Optional[bool] = None,
+    dialect: t.Optional[DialectType] = None,
```

Comment:
delete this, the dialect is already a part of schema

Reply:
If `schema = None` then we set `schema = sqlglot.schema` which doesn't have the same dialect as the `optimize()` step right?

Reply:
Do we actually still need `sqlglot.schema`? Not sure what's the context behind that, but IIRC it was added before implementing `ensure_schema`, so it may be redundant now.

Reply:
Removed it altogether